src/primitives.jl

```diff
@@ -212,17 +212,51 @@ end
 # There is no trait and this is not an AbstractArray.
 # Try to call _apply over it. We can't use threading
 # as we don't know if we can can index into it. So just `map`.
-@inline function _apply(f::F, target, ::Nothing, iterable; threaded, kw...) where F
-    if threaded
-        # `collect` first so we can use threads
-        _apply(f, target, collect(iterable); threaded, kw...)
-    else
-        apply_to_iterable(x) = _apply(f, target, x; kw...)
-        map(apply_to_iterable, iterable)
+@inline function _apply(f::F, target, ::Nothing, iterable::IterableType; threaded, kw...) where {F, IterableType}
+    # Try the Tables.jl interface first
+    if Tables.istable(iterable)
+        _apply_to_table(f, target, iterable; threaded, kw...)
+    else # this is probably some form of iterable...
+        if threaded isa _True
+            # `collect` first so we can use threads
+            _apply(f, target, collect(iterable); threaded, kw...)
+        else
+            apply_to_iterable(x) = _apply(f, target, x; kw...)
+            map(apply_to_iterable, iterable)
+        end
     end
 end
+#= 
+Doing this inline in `_apply` is _heavily_ type unstable, so it's best to separate this 
+by a function barrier.
+
+This function operates `apply` on the `geometry` column of the table, and returns a new table
+with the same schema, but with the new geometry column.
+
+This new table may be of the same type as the old one iff `Tables.materializer` is defined for 
+that table.  If not, then a `NamedTuple` is returned.
+=#
+function _apply_to_table(f::F, target, iterable::IterableType; threaded, kw...) where {F, IterableType}
+    _get_col_pair(colname) = colname => Tables.getcolumn(iterable, colname)
+    # We extract the geometry column and run `apply` on it.
+    geometry_column = first(GI.geometrycolumns(iterable))
+    new_geometry = _apply(f, target, Tables.getcolumn(iterable, geometry_column); threaded, kw...)
+    # Then, we obtain the schema of the table,
+    old_schema = Tables.schema(iterable)
+    # filter the geometry column out,
+    new_names = Iterators.filter(Base.Fix1(!==, geometry_column), old_schema.names)
+    # and try to rebuild the same table as the best type - either the original type of `iterable`,
+    # or a named tuple which is the default fallback.
+    return Tables.materializer(iterable)(
+        merge(
```

Comment:
I'm worried about compile time for this when there are 100+ columns. But its ok for now.

Reply:
Ah yeah...that might be an issue.

I could use a [Tables.DictColumnTable](https://tables.juliadata.org/stable/#Tables.dictcolumntable) instead, would that work better?

Reply:
Yeah that's probably better, its more like a DataFrame

Reply:
Allthough that does column-unioning which can be slow for other reasons...

Reply:
I guess I could manually pass in a Tables.Schema to solve the issue...